api/src/main/java/ca/bc/gov/educ/api/gradstudent/service/GradStudentReportService.java

```diff
@@ -51,37 +55,54 @@ public List<ReportGradStudentData> getGradStudentDataForNonGradYearEndReport() {
         return processReportGradStudentDataList(reportGradStudentDataPage);
     }
 
-    public List<ReportGradStudentData> getGradStudentDataForNonGradYearEndReport(String mincode) {
+    public List<ReportGradStudentData> getGradStudentDataForNonGradYearEndReportBySchool(UUID schoolId) {
         PageRequest nextPage = PageRequest.of(0, PAGE_SIZE);
-        if(StringUtils.isBlank(mincode)) {
-            throw new IllegalArgumentException("Invalid mincode: " + mincode);
+        if(schoolId == null) {
+            throw new IllegalArgumentException("Invalid schoolId: " + schoolId);
         }
         Page<ReportGradStudentDataEntity> reportGradStudentDataPage;
-        if(mincode.length() == 3) {
-            reportGradStudentDataPage = reportGradStudentDataRepository.findReportGradStudentDataEntityByDistcodeAndProgramCompletionDateAndStudentStatusAndStudentGrade(mincode, nextPage);
-        } else {
-            reportGradStudentDataPage = reportGradStudentDataRepository.findReportGradStudentDataEntityByMincodeAndProgramCompletionDateAndStudentStatusAndStudentGrade(mincode, nextPage);
+        reportGradStudentDataPage = reportGradStudentDataRepository.findReportGradStudentDataEntityBySchoolOfRecordIdAndProgramCompletionDateAndStudentStatusAndStudentGrade(schoolId, nextPage);
+        return processReportGradStudentDataList(schoolId, null, reportGradStudentDataPage);
+    }
+
+    public List<ReportGradStudentData> getGradStudentDataForNonGradYearEndReportByDistrict(UUID districtId) {
+        PageRequest nextPage = PageRequest.of(0, PAGE_SIZE);
```

Comment:
If there are more than PAGE_SIZE they will never be included?

Reply:
Yes, it is retrieving students for one district, and that will be huge volume of data.